core-js/src/test/javascript/cdf/components/UnmanagedComponent-spec.js

```diff
@@ -219,10 +222,85 @@ define([
     });
 
     /**
+     * ## Unmanaged Component # should allow changes in the datasources
+     */
+    it("should allow changes in the datasources", function(done) {
+      spyOn(uquery.dashboard,'isValidQueryDefinition').and.returnValue(true);
+      uquery.preExecution = function () {
+        this.queryDefinition.dataSource = "otherDatasource";
+      };
+      uquery._setQuery = function (queryDef, queryOptions) {
+        expect(queryDef.dataSource).toBe("otherDatasource");
+        done();
+      };
+      dashboard.update(uquery);
+    });
+
+    /**
+     * ## Unmanaged Component # should allow changes in the datasources by override
+     */
+    it("should allow changes  in the datasources by override", function(done) {
+      spyOn(uquery.dashboard,'isValidQueryDefinition').and.returnValue(true);
+      uquery.preExecution = function () {
+        this.queryDefinition = $.extend( {}, this.queryDefinition , { dataSource: "otherDatasource"});
+      };
+      uquery._setQuery = function (queryDef, queryOptions) {
+        expect(queryDef.dataSource).toBe("otherDatasource");
+        done();
+      };
+      dashboard.update(uquery);
+     });
+
+    /**
+     * ## Unmanaged Component # should allow changes in the datasources by merge
+     */
+    it("should allow changes datasources by  merge", function(done) {
```

Comment:
This test is the same as the [first one](https://github.com/webdetails/cdf/pull/804/files#diff-d2c12d404257fe84be9ede17fa383f33R227).